server/stream.go

```diff
@@ -1044,28 +1076,10 @@ func (s *Server) checkStreamCfg(config *StreamConfig, acc *Account) (StreamConfi
 		return exists, cfg.MaxMsgSize, cfg.Subjects
 	}
 
-	hasFilterSubjectOverlap := func(filter string, streamSubs []string) bool {
```

Comment:
Currently we do not allow colliding subjects in same account.

Reply:
Same argument as above for `validFilteredSubject()` this test is IMHO no longer valid because the stream you want to mirror (or source from) may contain messages with transformed subjects that do not match the stream's subscription subject filters (which are applied before the input subject transform), because a stream's filter can be changed after the stream's creation and because it prevents 'source chaining' between streams.

Reply:
Yes but within an account, you can not have two streams listening on `foo`, which would produce two pubAcks.

We need to maintain that restriction for now.

Reply:
That restriction is still there and active (only one stream can listen on a particular subject) and it still returns an error if you try to define a stream who's listen subject(s) overlap with another existing stream's listen subject(s).

This is a different check (only applies when creating a stream that mirrors another stream or sources from another stream(s), does not apply to what the stream is listening on)

Reply:
ok I will take another look to make sure, I thought you removed test and the check here as well.

Reply:
For reference the test for listen subject overlap doesn't happen in Server.checkStreamCfg() but instead happens in Account.addStreamWithAssignment() and in stream.updateWithAdvisory() using the jsAccount.subjectsOverlap() function.